pilot/pkg/model/service.go

```diff
@@ -619,16 +619,19 @@ func (s *Service) GetServiceAddressForProxy(node *Proxy) string {
 	return s.Address
 }
 
-// GetTLSModeFromEndpointLabels returns the value of the label
-// security.istio.io/tlsMode if set. Do not return Enums or constants
-// from this function as users could provide values other than istio/disabled
-// and apply custom transport socket matchers here.
-func GetTLSModeFromEndpointLabels(labels map[string]string) string {
+// GetTLSMode returns the tls mode applies to a service instance
+// 1. the value of the label "security.istio.io/tlsMode" if set.
+// 2. "istio" mTLS if service account is present.
+// 3. "disabled" if both the above mismatched.
+func GetTLSMode(labels map[string]string, serviceaccount string) string {
```

Comment:
@ramaraochavali This is not right,  the ci failed. I need to revert this

Reply:
Could it be because you have not put that logic in else if

```
} else if wle.ServiceAccount != "" {
				tlsMode = model.IstioMutualTLSModeLabel
			}
```